### PR TITLE
View competitions

### PR DESCRIPTION
**Pull request by mthembbu**

### CODE REVIEW

File: skietbaan_fe/src/actions/competition.action.js [hunk 1]
```diff
@@ -1,30 +1,51 @@
...
-/**A method to fetch all competitions */
-/**GC TODO: The method to fetch data from the  competition table 
-export const fetchComps = () => {
-	fetch('https://jsonplaceholder.typicalcode.com/posts')
-		.then((res) => res.json())
-		.then((comps) => dispatch({ type: FETCH_COMP, payload: comps }))
-};END OF GC*/
-/** Amethod to create a new competition and posts comps data to the designated url
- * TO DO: don't forget to use the http://skietbaan.retrotest.co.za/api/
+import { NEW_COMP, FETCH_COMP, UPDATE_COMP_STATUS,URL,LOCAL_URL } from './types';
```
Comment (FredJEhlers) [inline skietbaan_fe/src/actions/competition.action.js]: each import on its own line

File: skietbaan_fe/src/actions/competition.action.js [hunk 2]
```diff
@@ -1,30 +1,51 @@
...
+export const fetchcomp = () => (dispatch) => {
+	fetch(URL+"/api/Competition/all").then((res) => res.json()).then((compData) => {
+	//debugger;	
+	dispatch({
+			type: FETCH_COMP,
```
Comment (FredJEhlers) [inline skietbaan_fe/src/actions/competition.action.js]: fix indentation

File: skietbaan_fe/src/actions/competition.action.js [hunk 3]
```diff
@@ -1,30 +1,51 @@
...
  */
 export const createcomp = (compData) => (dispatch) => {
-	fetch('http://localhost:63474/api/Competition', {
+	fetch(URL+"/api/competition", {
 		method: 'POST',
 		headers: {
-			'Accept': 'application/json',
-          'Content-Type': 'application/json'
+			Accept: 'application/json',
+			'Content-Type': 'application/json'
```
Comment (FredJEhlers) [inline skietbaan_fe/src/actions/competition.action.js]: Accept does not have `' '` so either add them or remove content-type's `' `'`

File: skietbaan_fe/src/actions/types.js [hunk 4]
```diff
@@ -3,9 +3,11 @@ export const DELETE_BOOKMARK = 'DELETE_BOOKMARK';
...
 export const NEW_COMP = 'NEW_COMP';
-export const FETCH_COMP = 'NEW_COMP';
+export const FETCH_COMP = 'FETCH_COMP';
+export const UPDATE_COMP_STATUS = 'UPDATE_COMP_STATUS';
 export const BASE_URL = "https://api.skietbaan.retrotest.co.za";
 //leaderboard types
 export const FETCH_LEADERBOARDFILTER_DATA = 'FETCH_LEADERBOARDFILTER_DATA';
 export const FETCH_LEADERBOARDTABLE_DATA = 'FETCH_LEADERBOARDTABLE_DATA';
 export const URL = "https://api.skietbaan.retrotest.co.za"; 
+export const LOCAL_URL = "http://localhost:63474";
```
Comment (FredJEhlers) [inline skietbaan_fe/src/actions/types.js]: No localhost variables

File: skietbaan_fe/src/actions/competition.action.js [hunk 5]
```diff
@@ -1,30 +1,51 @@
...
+export const updateByIDcomp = (compData, Id) => (dispatch) => {
+	fetch(URL+"/api/Competition/" + Id, {
+		method: 'PUT',
+		headers: {
+			Accept: 'application/json',
+			'Content-Type': 'application/json'
+		},
+		body: JSON.stringify(compData)
+	})
+		.then((res) => {res.json(); console.log(res)})
```
Comment (FredJEhlers) [inline skietbaan_fe/src/actions/competition.action.js]: move `.then` to line 41 so its `}),then(`
Also remove logs

File: skietbaan_fe/src/components/CreateComp.js [hunk 6]
```diff
@@ -1,63 +1,101 @@
...
 import '../scss/createcomp.css';
 import { createcomp } from '../actions/competition.action';
-//import axios from 'axios';
 class CreateComp extends Component {
-	constructor() {
-		super();
-		this.state = { Name: '', Status: true };
+	constructor(props) {
+		super(props);
+		this.state = { Name: '', BestScoresNumber: 0, Status: true,
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/CreateComp.js]: each entry on its own line

File: skietbaan_fe/src/components/CreateComp.js [hunk 7]
```diff
@@ -1,63 +1,101 @@
...
+		this.onClick = this.onClick.bind(this);
+	}
+	onClick() {
+		Window.location = '/create';
 	}
 	/** A method that detects the change in the change in thw textfield */
 	onChange(e) {
 		this.setState({ [e.target.name]: e.target.value });
 	}
+	onChangeInt(e) {
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/CreateComp.js]: Space between declaration

File: skietbaan_fe/src/components/ViewComp.js [hunk 8]
```diff
@@ -1,16 +1,108 @@
...
+		super(props);
+		this.state = {
+			allComps: [],
+			status: [],
+			updatedComp: {}
+		};
+		this.changeActive = this.changeStatus.bind(this);
+		this.onClick = this.onClick.bind(this);
+	}
+	onClick(){
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/ViewComp.js]: Space between declarations

File: skietbaan_fe/src/reducers/createcomp.reducer.js [hunk 9]
```diff
@@ -1,7 +1,8 @@
-import {NEW_COMP, FETCH_COMP} from '../actions/types';
+import {NEW_COMP, FETCH_COMP, UPDATE_COMP_STATUS} from '../actions/types';
```
Comment (FredJEhlers) [inline skietbaan_fe/src/reducers/createcomp.reducer.js]: each import on its own line

File: skietbaan_fe/src/scss/view-comp.css [hunk 10]
```diff
@@ -0,0 +1,67 @@
+.activeButton button {
+	background-color: #e11e2b;
+    border-radius: 20px;
+    border: none;
+	color: black;
+	width: 116px;
+    height: 42px;
+    font-weight: lighter;
+}
+.inactiveButton button {
```
Comment (FredJEhlers) [inline skietbaan_fe/src/scss/view-comp.css]: Space between declartions

File: skietbaan_fe/src/scss/view-comp.css [hunk 11]
```diff
@@ -0,0 +1,67 @@
+.activeButton button {
+	background-color: #e11e2b;
+    border-radius: 20px;
```
Comment (FredJEhlers) [inline skietbaan_fe/src/scss/view-comp.css]: inconsistent indentation

Comment (FredJEhlers) [reply skietbaan_fe/src/scss/view-comp.css]: Throughout the file

File: skietbaan_fe/src/components/CreateComp.js (hunk 7 above)
Comment (mthembbu) [reply skietbaan_fe/src/components/CreateComp.js]: Hi Fred, please clarify what you mean by a space between declaration

File: skietbaan_fe/src/scss/view-comp.css (hunk 11 above)
Comment (mthembbu) [reply skietbaan_fe/src/scss/view-comp.css]: But I right-clicked -> format document?

Comment (FredJEhlers) [reply skietbaan_fe/src/scss/view-comp.css]: It didnt work 🤷‍♂️

File: skietbaan_fe/src/components/CreateComp.js (hunk 7 above)
Comment (FredJEhlers) [reply skietbaan_fe/src/components/CreateComp.js]: spaces between functions 
```
function cookies(){}
// the space I was refering to 
function eat(){}
```

**Pull request merged**